tests/components/timer/test_init.py

```diff
@@ -342,14 +342,14 @@ async def test_start_service(hass: HomeAssistant) -> None:
     await hass.services.async_call(
         DOMAIN,
         SERVICE_CHANGE,
-        {CONF_ENTITY_ID: "timer.test1", CONF_DURATION: -3},
+        {CONF_ENTITY_ID: "timer.test1", CONF_DURATION: -7},
```

Comment:
This change is not correct, it hides a bug introduced by this PR.
Instead, `async_change` needs to be modified to compare the change with running duration, for example like this:
```diff
diff --git a/homeassistant/components/timer/__init__.py b/homeassistant/components/timer/__init__.py
index 236e3c99091..2c2e6b79d6c 100644
--- a/homeassistant/components/timer/__init__.py
+++ b/homeassistant/components/timer/__init__.py
@@ -317,11 +317,11 @@ class Timer(collection.CollectionEntity, RestoreEntity):
         self._state = STATUS_ACTIVE
         start = dt_util.utcnow().replace(microsecond=0)

-        # Set remaining to new value if needed
+        # Set remaining and running duration unless resuming or restarting
         if duration:
             self._remaining = self._running_duration = duration
         elif not self._remaining:
-            self._remaining = self._duration
+            self._remaining = self._running_duration = self._duration

         self._end = start + self._remaining

@@ -339,7 +339,7 @@ class Timer(collection.CollectionEntity, RestoreEntity):
             raise HomeAssistantError(
                 f"Timer {self.entity_id} is not running, only active timers can be changed"
             )
-        if self._remaining and (self._remaining + duration) > self._duration:
+        if self._remaining and (self._remaining + duration) > self._running_duration:
             raise HomeAssistantError(
                 f"Not possible to change timer {self.entity_id} beyond configured duration"
             )
```

Reply:
Fine to make the other changes but the testcase is still correct?!
If you have 15 seconds off duration and the subtract with 7 it becomes 8 so therefore it's within the duration (actually both configured and running)

Reply:
The reason you previously had to change this was to hide the regression which is fixed by the diff above; adjusting by -3 seconds was not possible when the duration was compared to 10s (wrong), but possible when compared to 15s.
Please revert it.
```suggestion
        {CONF_ENTITY_ID: "timer.test1", CONF_DURATION: -3},
```